pm/queue.go

```diff
@@ -110,22 +95,41 @@ func (q *ticketQueue) startQueueLoop() {
 	sub := q.blockSub(blockNums)
 	defer sub.Unsubscribe()
 
+ticketLoop:
 	for {
 		select {
 		case err := <-sub.Err():
 			if err != nil {
 				glog.Errorf("Block subscription error err=%v", err)
 			}
 		case latestBlock := <-blockNums:
-			numTickets := q.Length()
+			numTickets, err := q.Length()
+			if err != nil {
+				glog.Errorf("Error getting queue length err=%v", err)
+				continue
+			}
 			for i := 0; i < int(numTickets); i++ {
-				q.cond.L.Lock()
-				nextTicket := q.queue[0]
-				q.cond.L.Unlock()
-				q.removeHead()
+				nextTicket, err := q.pop()
```

Comment:
Seems like this could be replaced with:

```golang
nextTicket, err := q.store.SelectEarliestWinningTicket(q.sender)
```

Reply:
fixed in 985649238628b9e8d824d2c54280bd2e84da0466